app/containers/markdown/Hashtag.tsx

```diff
@@ -15,15 +21,27 @@ interface IHashtag {
 
 const Hashtag = React.memo(({ hashtag, channels, navToRoomInfo, style = [] }: IHashtag) => {
 	const { theme } = useTheme();
+	const isMasterDetail = useAppSelector(state => state.app.isMasterDetail);
+	const navigation = useNavigation<StackNavigationProp<ChatsStackParamList, 'RoomView'>>();
 
-	const handlePress = () => {
+	const handlePress = async () => {
 		const index = channels?.findIndex(channel => channel.name === hashtag);
 		if (typeof index !== 'undefined' && navToRoomInfo) {
 			const navParam = {
 				t: 'c',
 				rid: channels?.[index]._id
 			};
-			navToRoomInfo(navParam);
+			const room = await getSubscriptionByRoomId(navParam.rid!);
+			if (room) {
+				if (isMasterDetail) {
+					// Close the modal if it is open and then redirect to the channel
+					Navigation.navigate('DrawerNavigator');
+					navigation.replace('RoomView', room);
```

Comment:
Maybe you should use `goRoom`.
It's going to be hard to maintain this in case we change `goRoom` logic.